pykechain/models/widgets/widget.py

```diff
@@ -166,20 +155,27 @@ def associated_parts(self, *args, **kwargs):
         :returns: a tuple(models of :class:`PartSet`, instances of :class:`PartSet`)
 
         """
-        return (
-            self.parts(category=Category.MODEL, *args, **kwargs),
-            self.parts(category=Category.INSTANCE, *args, **kwargs)
-        )
+        models_and_instances = self._client.parts(*args, widget=self.id, category=None, **kwargs)
+
+        models = [p for p in models_and_instances if p.category == Category.MODEL]
+        instances = [p for p in models_and_instances if p.category == Category.INSTANCE]
+
+        return models, instances
 
     #
     # Write methods
     #
-    def update_associations(self, readable_models=None, writable_models=None, **kwargs):
-        # type: (Optional[List], Optional[List], **Any) -> None
+
+    def update_associations(
```

Comment:
ref to the other comment:

add part_instance and parent_part_instance as options to set the associations in the backend.